c2rust-analyze/src/known_fn.rs

```diff
@@ -602,28 +622,823 @@ pub const fn all_known_fns() -> &'static [KnownFn] {
             ) -> c_int;
 
             fn access(
-                path: *const c_char: [READ | OFFSET_ADD],
+                path: *const c_char: [READ | OFFSET_ADD | NON_NULL],
                 amode: c_int,
             ) -> c_int;
 
+            fn alarm(
+                seconds: c_uint,
+            ) -> c_uint;
+
+            fn atoi(
+                s: *const c_char: [READ | OFFSET_ADD | NON_NULL],
+            ) -> c_int;
+
+            fn bind(
+                socket: c_int,
+                address: *const sockaddr: [READ],
+                address_len: socklen_t,
+            ) -> c_int;
+
+            fn calloc(
+                nobj: size_t,
+                size: size_t,
+            ) -> *mut c_void: [READ | WRITE | OFFSET_ADD | FREE];
+
+            fn chdir(
+                dir: *const c_char: [READ | OFFSET_ADD | NON_NULL],
```

Comment:
Might be nice to have an alias for the "standard" string-argument permissions, since they show up in so many functions.

Reply:
Should `NON_NULL` be included in that?  I'd imagine not, but then it's not saving a ton over just writing `READ | OFFSET_ADD`.  And `READ | OFFSET_ADD` could just be `SLICE` to be more generic than `STR`.

Reply:
I was thinking `READ | OFFSET_ADD | NON_NULL`, since that seems to be the most common case

Reply:
I'm not sure if it's worth it to do this.  If I do `STR = READ | OFFSET_ADD | NON_NULL`, then for a nullable string, I'd have to do `READ | OFFSET_ADD`, which I think is confusing.  And it's not saving so much repetition; we're only saving a max of 2 permissions per ptr.

If we ever add something like `NULL_TERMINATED` as a flag or something, then it would make more sense in that case.